drivers/serial/uart_aesc.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025 aesc silicon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aesc_uart
+
+#include <errno.h>
+#include <soc.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/aesc/ip_identification.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(aesc_uart, CONFIG_UART_LOG_LEVEL);
+
+#define DEV_UART_IRQ_TX_EN		(1 << 0)
+#define DEV_UART_IRQ_RX_EN		(1 << 1)
+
+struct uart_aesc_data {
+	DEVICE_MMIO_NAMED_RAM(regs);
+};
+
+struct uart_aesc_config {
+	DEVICE_MMIO_NAMED_ROM(regs);
+	uint64_t sys_clk_freq;
+	uint32_t current_speed;
+};
+
+struct uart_aesc_regs {
+	uint32_t data_width;
+	uint32_t sampling_sizes;
+	uint32_t fifo_depths;
+	uint32_t permissions;
+	uint32_t read_write;
+	uint32_t status;
+	uint32_t clock_div;
+	uint32_t frame_cfg;
+	uint32_t ip;
+	uint32_t ie;
+};
+
+#define DEV_CFG(dev) ((struct uart_aesc_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct uart_aesc_data *)(dev)->data)
+#define DEV_UART(dev)							     \
+	((struct uart_aesc_regs *)DEVICE_MMIO_NAMED_GET(dev, regs))
+
+
+static void uart_aesc_poll_out(const struct device *dev, unsigned char c)
+{
+	volatile struct uart_aesc_regs *uart = DEV_UART(dev);
+
+	while ((uart->status & 0x00FF0000) == 0) {
+	}
+	uart->read_write = c;
+}
+
+static int uart_aesc_poll_in(const struct device *dev, unsigned char *c)
+{
+	volatile struct uart_aesc_regs *uart = DEV_UART(dev);
+	int val;
+
+	val = uart->read_write;
+	if (val & 0x10000) {
```

Comment:
Assign this magic number to a well-named variable, and use the variable instead.